src/main/java/com/xpanxion/java/springboot/da1/demo/service/student1/GymService1.java

```diff
@@ -25,9 +28,26 @@
     @Autowired
     private GymMemberRepository1 gymMemberRepository1;
 
+    @Autowired
+    private WorkoutHistoryRepository1 workoutHistoryRepository1;
+
     @Autowired
     private TimestampsRepository1 timestampsRepository1;
 
+    WorkoutHistory1 workoutHistory1;
+
+    public List<WorkoutHistory1> getWorkoutHistory(Integer memberId) {
+
+        var workoutHistory = workoutHistoryRepository1.findByMemberId(memberId);
+
+        if (workoutHistory.size() == 0 ) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND);
```

Comment:
In general, ReST APIs do not throw a 404 when a resource doesn't have any data.  A better code might be a 204 NO_CONTENT.

Reply:
Just pushed this correction